src/main/java/net/fabricmc/tinyremapper/ClassInstance.java

```diff
@@ -174,19 +174,22 @@ public String getSignature() {
 	}
 
 	@Override
-	public MemberHeader getField(String name, String desc) {
+	public TrMember resolveField(String name, String desc) {
 		return this.resolve(MemberType.FIELD, MemberInstance.getFieldId(name, desc, desc == null));
 	}
```

Comment:
The internal implementation uses `tr.ignoreFieldDesc` everywhere. Say if the `tr.ignoreFieldDesc = true` and someone passes `null`, then it will always return no match since the internal id does not match (one id has desc another id does not). Vice versa.

Reply:
Here is code that make things work

```Java
	@Override
	public TrMember resolveField(String name, String desc) {
		if (desc == null) {
			return this.resolvePartial(MemberType.FIELD, name, ";;");
		} else {
			return this.resolve(MemberType.FIELD, MemberInstance.getFieldId(name, desc, tr.ignoreFieldDesc));
		}
	}
```